rinad/src/ipcm/ipcm.cc

```diff
@@ -395,6 +400,26 @@ IPCManager_::assign_to_dif(Addon* callee, Promise* promise,
 	IPCPTransState* trans;
 
 	try {
+		if (is_any_ipcp_assigned_to_dif(dif_name)) {
+			ss << "There is already an IPCP assigned to DIF "
+				<< dif_name.toString()
+				<< " in this system.";
+			FLUSH_LOG(ERR, ss);
+			throw rina::AssignToDIFException();
+		}
+
+		ipcp = lookup_ipcp_by_id(ipcp_id, false);
+		if (ipcp->get_type() == rina::NORMAL_IPC_PROCESS) {
+			// Load all the plugins required from by template, but
+			// first release the ipcp lock, to avoid lock-ups.
+			ipcp->rwlock.unlock();
+
+			catalog.load_by_template(callee, ipcp_id, dif_template);
```

Comment:
I don't fully understand the rationale here. I think we should clearly separate the template manager from the catalog.

At first glance, I would say that the IPCM should be the only one who should "interact or bridge" between these two modules => do not embbed template logic stuff in the catalogue and viceversa.

In this case, what I think it should be done is that the IPCM recovers via the template (or whatever) the list of policies, and _then_ it actually loads them, making catalogue unaware of templates.